app/parser/v0_0_1/schema_parser.py

```diff
@@ -81,190 +37,67 @@ def get_parser_version(self):
     def parse(self):
         """Parse the schema
 
-        :returns: A questionnaire object
-
         :raises: A SchemaParserException if there is a problem while parsing the schema
 
         """
-        questionnaire = Questionnaire()
-        questionnaire.id = ParserUtils.get_required_string(self._schema, "questionnaire_id")
-        questionnaire.title = ParserUtils.get_required_string(self._schema, "title")
-        questionnaire.survey_id = ParserUtils.get_required_string(self._schema, "survey_id")
-        questionnaire.description = ParserUtils.get_optional_string(self._schema, "description")
-        questionnaire.theme = ParserUtils.get_required_string(self._schema, "theme")
-        questionnaire.data_version = ParserUtils.get_required_string(self._schema, "data_version")
-
-        if "introduction" in self._schema.keys():
-            questionnaire.introduction = self._parse_introduction(self._schema['introduction'])
-
         if "groups" in self._schema.keys():
             for group_schema in self._schema['groups']:
-                questionnaire.add_group(self._parse_group(group_schema, questionnaire))
+                self._parse_group(group_schema)
         else:
             raise SchemaParserException('Questionnaire must contain at least one group')
```

Comment:
Yes, it's difficult. I've attempted to do it by making all arrays have a minimum of 1 item in json schema. The resultant problem becomes that interstitials currently define empty lists for answers/questions and all the schemas have empty arrays to suppress the exceptions raised by this schema parser. I'd rather save it for another PR, given this is simply getting WTForms in.